encodings/fastlanes/src/for/compute/compare.rs

```diff
@@ -53,10 +51,15 @@ where
         if let Some(reference) = reference {
             rhs = rhs.wrapping_sub(&reference);
         }
-        if lhs.shift() > 0 {
-            // Since compare requires that both sides are of same dtype this will always succeed and not panic
-            rhs = rhs >> (lhs.shift() as u32)
-        }
+        // Since compare requires that both sides are of same dtype this will always succeed and not panic
+
+        let zero = T::zero();
+        let rhs = match rhs.checked_shr(lhs.shift() as u32) {
```

Comment:
Can we just use `if lhs.shift() == T::bit_width() { if rhs == T::zero() { ... } else { vortex_panic() }` and handle the case explicitly?